cmd/internal/genopenapi3/gen_converter.go

```diff
@@ -0,0 +1,93 @@
+package genopenapi3
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"go.aporeto.io/regolithe/spec"
+)
+
+const paramNameID = "id"
+
+type converter struct {
+	inSpecSet  spec.SpecificationSet
+	outRootDoc openapi3.T
+}
+
+func newConverter(inSpecSet spec.SpecificationSet) *converter {
+	specConfig := inSpecSet.Configuration()
+	c := &converter{
+		inSpecSet: inSpecSet,
+		outRootDoc: openapi3.T{
+			OpenAPI: "3.0.3",
+			Info: &openapi3.Info{
+				Title:          specConfig.Name,
+				Version:        specConfig.Version,
+				Description:    specConfig.Description,
+				TermsOfService: specConfig.Copyright + " © (TODO)", // TODO
+				License: &openapi3.License{
+					Name: "TODO",
+				},
+				Contact: &openapi3.Contact{
+					Name:  specConfig.Author,
+					URL:   specConfig.URL,
+					Email: specConfig.Email,
+				},
+			},
+			Paths: openapi3.Paths{},
+			Components: openapi3.Components{
+				Schemas: make(openapi3.Schemas),
+			},
+		},
+	}
+	return c
+}
+
+func (c *converter) Do(dest io.Writer) error {
+
+	for _, s := range c.inSpecSet.Specifications() {
+		if err := c.processSpec(s); err != nil {
+			return fmt.Errorf("unable to to process spec: %w", err)
```

Comment:
to avoid feature envy by penetrating into the model at the outer scope, this information is included near where we fail to convert the model as follows:

```
func (c *converter) processSpec(s spec.Specification) error {
....
	schema, err := c.convertModel(s)
	if err != nil {
		return fmt.Errorf("model '%s': %w", s.Model().RestName, err)
	}
...
}
```

this is an example of an error:
> error generating openapi3 document from spec set 'gaia': unable to to process spec: model 'timeseriesrow': attribute 'values': retrieving 'openapi3' type mapping for external attribute subtype '[][]interface{}': no mode 'openapi3' found in type mapping